Sources/Build/BuildOperationBuildSystemDelegateHandler.swift

```diff
@@ -109,31 +109,36 @@ final class TestDiscoveryCommand: CustomLLBuildCommand, TestBuildCommand {
     }
 
     private func execute(fileSystem: Basics.FileSystem, tool: TestDiscoveryTool) throws {
+        let outputs = tool.outputs.compactMap { try? AbsolutePath(validating: $0.name) }
+
+        if self.context.buildParameters.testingParameters.library == .swiftTesting {
+            for file in outputs {
+                try fileSystem.writeFileContents(file, string: "")
```

Comment:
seems a bit unfortunate?

Reply:
Teaching the various layers that these files aren't necessary for swift-testing proved harder than my brain could handle solo. If we can revise the relevant middle logic, I'm on board with getting rid of this code.

Reply:
So for swift-testing, we don't actually care about the test target's object files?

Reply:
All we need for swift-testing is the entrypoint for a target; we don't need nor perform any individual discovery. As long as everything gets linked into the binary (or is loaded at runtime), it's visible to swift-testing.

Reply:
OK, let me think about it and I can make some suggestions how to handle that difference.

Reply:
I think we can change `addTestDiscoveryGenerationCommand()` instead? We need to pass down the swift-testing bit and if it is true, we would adjust the generated command accordingly. Not sure if you tried that already and there are complications which I am missing.

Reply:
I did try messing with that function first, but it fails and I ended up down a rabbit hole:

> error: couldn't build PATH/arm64-apple-macosx/debug/swift_testingPackageDiscoveredTests.build/sources because of missing inputs: PATH/.build/arm64-apple-macosx/debug/swift-testingPackageDiscoveredTests.derived/TestingTests.swift, PATH/.build/arm64-apple-macosx/debug/swift-testingPackageDiscoveredTests.derived/all-discovered-tests.swift, PATH/.build/arm64-apple-macosx/debug/swift-testingPackageDiscoveredTests.derived/TestingMacrosTests.swift

What I've got here is obviously not perfect but it compiles and runs, dag nabbit. 🙃 I was hoping we might proceed with it for now, but revise once you've had time to page in the relevant code and I've had a chance to learn more about the layering.

Reply:
Maybe if we switch to `writeIfChanged()` this could be acceptable? I think right now, I'm pretty sure it'll mess with incremental builds since we'll always be writing those files.

Reply:
Sure, I can do that.

Reply:
Done.